src/util/hash.c

```diff
@@ -124,7 +122,6 @@ pmix_status_t pmix_hash_fetch(pmix_hash_table_t *table, int rank,
         /* specified rank can return following statuses
          * PMIX_ERR_PROC_ENTRY_NOT_FOUND | PMIX_ERR_NOT_FOUND | PMIX_SUCCESS
          * special logic is basing on these statuses on a client and a server */
-        id = (uint64_t)rank;
```

Comment:
If we remove this line, then we can just remove the entire "else" clause, can't we?